### ISSUE TITLE
[DOCS] http.query isn't documented effectively

### ISSUE THREAD

**Issue by OrangeDog** [saltstack/salt#59930]
**Description**
The documentation for the [http module](https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.http.html) doesn't explain any of the 49 parameters. Instead it tries to link to `utils.http.query`, which doesn't work.

**Suggested Fix**
Some options:
1. Copy the docstring from `utils.http.query` to `modules.http.query` and reverse the direction of the reference.
2. Get the documentation to generate for the `utils.http` modules and ensure the link works cortrectly.

**Type of documentation**
Salt modules

**Location or format of documentation**
https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.http.html

**Comment by OrangeDog** [comment]
To follow up on this, the doc inclusion does actually work, though the rendered link doesn't go anywhere.

The bigger problem is that the documentation for this 49-ary function is a single sentence:

>  Query a resource, and decode the return data

**Comment by sharkbruhaha** [comment]
Is there a WIP of the documentation or anything yet at least?

**Comment by barbaricyawps** [comment]
Not yet, but if you'd prefer to pick it up, I can remove myself as an assignee.

**Comment by sharkbruhaha** [comment]
no, just stumbled on this and saw the date, so was wondering as I struggled through figuring it out.

(also, it _seems_ like half the options don't actually have any affect, assuming that my tested values weren't wrong just because no error thrown)

**Comment by mdschmitt** [comment]
Anybody on this?  I'd love to use http.query more effectively.

**Comment by OrangeDog** [comment]
This has not been done. Only one parameter (`raise_error`) is documented.
https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.http.html#salt.utils.http.query
